src/main/antlr/Asciidoc.g4

```diff
@@ -1,10 +1,53 @@
 grammar Asciidoc;
 
 @header {
-    package org.asciidoctor.parser;
+package org.asciidoctor.parser;
 }
 
-// TODO replace with actual AsciiDoc grammar!
-r  : 'hello' ID ;
-ID : [a-z]+ ;
-WS : [ \t\r\n]+ -> skip ;
+document        : header preamble body ;
+
+// NOTE: "You cannot have a revision line without an author line."
+header          : document_title author_line? revision_line? attributes? NEW_LINE  ;
+
+document_title  : '=' phrase NEW_LINE ;
+author_line     : CHARS CHARS '<' email '>' NEW_LINE ;
+ // based on example v1.0, 2014-01-01
+revision_line   : 'v' NUMBERS '.' NUMBERS NEW_LINE ;
+attributes      : ':' '!'? atribute_name '!'? ':' atribute_value NEW_LINE ;
+atribute_name   : word ;
+atribute_value  : ANYCHAR ;
+
+preamble        : paragraph  ;
+paragraph       : phrase* NEW_LINE;
+
+body            : section* ;
+section         : section_header section_body ;
+section_header  : '=='+ phrase ;
```

Comment:
Section header could be odd number of the equal signs, I am not sure whether "==="  will match here.